texastribune/donations-app
# Dockerize

#### What's this PR do?
- Dockerizes the app.

#### Why are we doing this? How does it help us?
- So you don't have to worry about installing tools like `rbenv` and `aws-cli` on your machine
- Several of our other apps use Docker, so this keeps things consistent

#### How should this be manually tested?
- Set up an `env-docker` in the root and fill it out according to what's in `env.sample`
- `make`
- `yarn run dev`. **Note**: You're probably going to see a slew of warnings in the terminal. This is expected; they're all because the app uses the outdated Bourbon and Neat CSS libraries.
- Confirm you can view the app at `http://localhost:4567/index.html`
- Make a change to `source/index.html.erb`. Confirm the page automatically reloads and shows that change.
- Add a `console.log()` statement to `source/javascripts/index.js` and confirm the page automatically reloads and that the message is in the browser console.
- Add a selector to `source/scss/all.scss` and confirm the page automatically reloads and that the style change is reflected.
- Control+C out of the dev server.
- `yarn run build`
- Confirm Middleman builds the app.
- `yarn run deploy`. **Note**: The `--dryrun` flag is added to the deployment commands, so `aws-cli` will only simulate a deployment. I'll obviously remove that flag after this PR is reviewed.
- Confirm you get a series of messages about S3 syncing files.

#### Have automated tests been added?
No.

#### Are there performance implications? If adding JS can it be done async? Do images/CSS/JS have cache headers set?
No.

#### What are the relevant tickets?
Related to this one:
https://3.basecamp.com/3098728/buckets/736178/todos/673934858

#### How should this change be communicated to end users?
NA.

#### Next steps?
NA.

#### Smells?
I'll let @x110dc be the judge of that.

#### TODOs:
NA.

#### Has the relevant documentation/wiki been updated?
Yep, I updated the README.

#### Technical debt note
Less.
